alts/src/main/java/io/grpc/alts/AuthorizationUtil.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.alts;
+
+import com.google.common.collect.ImmutableList;
+import io.grpc.ServerCall;
+import io.grpc.Status;
+import io.grpc.alts.internal.AltsAuthContext;
+import io.grpc.alts.internal.AltsProtocolNegotiator;
+
+/** Utility class for ALTS client authorization. */
+public final class AuthorizationUtil {
+
+  private AuthorizationUtil() {}
+
+  /**
+   * Given a server call, performs client authorization check, i.e., checks if the client service
+   * account matches one of the expected service accounts.
+   */
+  public static Status clientAuthorizationCheck(
+      ServerCall<?, ?> call, ImmutableList<String> expectedServiceAccounts) {
+    AltsAuthContext altsCtx =
+        (AltsAuthContext) call.getAttributes().get(AltsProtocolNegotiator.AUTH_CONTEXT_KEY);
```

Comment:
Why isn't `AUTH_CONTEXT_KEY` a `Attributes.Key<AltsAuthContext>`?

Reply:
This because `extractPeerObject()` returns an Object, rather than AltAuthContext. Internally, we have a different AuthContext class. We need to use share the same API for returning the peer object. Let me know if you have better ideas.